src/main/java/de/cuuky/varo/game/start/ProtectionTime.java

```diff
@@ -20,9 +21,14 @@ public ProtectionTime(int timer) {
 	}
 
 	private void startGeneralTimer(int timer) {
-		this.scheduler = new BukkitRunnable() {
+		this.protectionTimer = timer;
 
-			private int protectionTimer = timer;
+		if (this.protectionTimer == 0) {
+			Main.getVaroGame().setProtection(null);
+			return;
+		}

```

Comment:
Why was this added?

Reply:
When I tried to use the remaining protection time in the Actionbar, there were a lot of error messages after the protection time was over, this seemed to fix it.

Reply:
That doesn't really make sense. `timer` would have to be 0 for this to do anything. But timer can't be 0 because that is already checked in `VaroGame`. So unless I am missing something that if statement does absolutely nothing

Reply:
> That doesn't really make sense. `timer` would have to be 0 for this to do anything. But timer can't be 0 because that is already checked in `VaroGame`. So unless I am missing something that if statement does absolutely nothing

This just fixed the issue where after the protection time it started spamming an error in the console. I would keep this if statement in just for safety.

Reply:
But it doesn't do anything unless you call the constructor from another plugin

Reply:
> But it shouldn't do anything unless you call the constructor from another plugin
> 
> > This just fixed the issue where after the protection time it started spamming an error in the console. I would keep this if statement in just for safety.
> 
> What was the error?

Can't reproduce the error right now but in my plugin, I'm using `Main.getVaroGame().getProtection() == null` for an if statement

Reply:
```suggestion
		if (this.protectionTimer == 0) {
			throw new IllegalArgumentException();
			return;
		}
```
This would be ok

But you probably had a different error somewhere else in your plugin

Reply:
> This would be ok
> 
> But you probably had a different error somewhere else in your plugin

I would just keep `Main.getVaroGame().setProtection(null);` in there. Before I had `Main.getVaroGame().setProtection(null);` in the Varo Plugin, it made the errors.

Reply:
Unless you can tell me exactly what this fixes, no.

As far as I can tell that code is never actually executed

Reply:
> Unless you can tell me exactly what this fixes, no.
> 
> As far as I can tell that code is never actually executed

I'll try reproducing the error

Reply:
> Unless you can tell me exactly what this fixes, no.
> 
> As far as I can tell that code is never actually executed

Okay so I tried reproducing the error but I couldn't get it to reproduce.
I'm using a Placeholder in the Actionbar and after the protection time is over, it goes to the placeholder text again so for example %placeholder% just in white. (I'm using an if `Main.getVaroGame().getProtection() != null` statement because I'm displaying the remaining ProtectionTime in my placeholder)

Reply:
> I'm using a Placeholder in the Actionbar and after the protection time is over, it goes to the placeholder text again so for example %placeholder% just in white. (I'm using an if `Main.getVaroGame().getProtection() != null` statement because I'm displaying the remaining ProtectionTime in my placeholder)

That has probably nothing to do with this. The varoplugin itself does not have a placeholder for the remaining protection time, so this is likely an issue with whatever plugin you are using for the placeholder or actionbar.

Reply:
> > I'm using a Placeholder in the Actionbar and after the protection time is over, it goes to the placeholder text again so for example %placeholder% just in white. (I'm using an if `Main.getVaroGame().getProtection() != null` statement because I'm displaying the remaining ProtectionTime in my placeholder)
> 
> That has probably nothing to do with this. The varoplugin itself does not have a placeholder for the remaining protection time, so this is likely an issue with whatever plugin you are using for the placeholder or actionbar.

Yeah, I'll just get `getCountdownSec` and `getCountdownMin` to check if both are zero